### PR TITLE
Add MI300X system tuning guide

### PR DESCRIPTION
**Pull request by peterjunpark**
This PR adds two areas of the MI300X tuning guide
https://github.com/ROCm/ROCm/pull/3416 and https://github.com/ROCm/ROCm/pull/3372 to `develop`.

### CODE REVIEW

File: docs/how-to/tuning-guides/mi300x.rst [hunk 1]
```diff
@@ -0,0 +1,1471 @@
...
+``70b``). 
+
+As described in the
+`AMD Instinct MI300X Accelerator <https://www.amd.com/content/dam/amd/en/documents/instinct-tech-docs/data-sheets/amd-instinct-mi300x-data-sheet.pdf>`__
+data sheet, the GPU memory capacity is 192 GB. This means you can run
+llama2-70b and llama3-70b models on one GPU.
+
+To maximize the accumulated throughput, you can also run eight instances
+vLLM simultaneously on one MI300X node (with eight GPUs). To do so, use
+the GPU isolation environment variable ``ROCR_VISIBLE_DEVICES``.
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:587]:
```suggestion
the GPU isolation environment variable ``CUDA_VISIBLE_DEVICES``.
```

File: docs/how-to/tuning-guides/mi300x.rst [hunk 2]
```diff
@@ -0,0 +1,1471 @@
...
+
+For example, this script runs eight instances of vLLM for throughput
+benchmarking at the same time:
+
+.. code-block:: shell
+
+   ROCR_VISIBLE_DEVICES="0" python3
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:594]:
```suggestion
   CUDA_VISIBLE_DEVICES="0" python3
```

File: docs/how-to/tuning-guides/mi300x.rst [hunk 3]
```diff
@@ -0,0 +1,1471 @@
...
+   /vllm-workspace/benchmarks/benchmark_throughput.py --dataset
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+   ROCR_VISIBLE_DEVICES="1" python3
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:599]:
```suggestion
   CUDA_VISIBLE_DEVICES="1" python3
```

File: docs/how-to/tuning-guides/mi300x.rst [hunk 4]
```diff
@@ -0,0 +1,1471 @@
...
+   /vllm-workspace/benchmarks/benchmark_throughput.py --dataset
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+   ROCR_VISIBLE_DEVICES="2" python3
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:604]:
```suggestion
   CUDA_VISIBLE_DEVICES="2" python3
```

File: docs/how-to/tuning-guides/mi300x.rst [hunk 5]
```diff
@@ -0,0 +1,1471 @@
...
+   /vllm-workspace/benchmarks/benchmark_throughput.py --dataset
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+   ROCR_VISIBLE_DEVICES="3" python3
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:609]:
```suggestion
   CUDA_VISIBLE_DEVICES="3" python3
```

File: docs/how-to/tuning-guides/mi300x.rst [hunk 6]
```diff
@@ -0,0 +1,1471 @@
...
+   /vllm-workspace/benchmarks/benchmark_throughput.py --dataset
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+   ROCR_VISIBLE_DEVICES="4" python3
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:614]:
```suggestion
   CUDA_VISIBLE_DEVICES="4" python3
```

File: docs/how-to/tuning-guides/mi300x.rst [hunk 7]
```diff
@@ -0,0 +1,1471 @@
...
+   /vllm-workspace/benchmarks/benchmark_throughput.py --dataset
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+   ROCR_VISIBLE_DEVICES="5" python3
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:619]:
```suggestion
   CUDA_VISIBLE_DEVICES="5" python3
```

File: docs/how-to/tuning-guides/mi300x.rst [hunk 8]
```diff
@@ -0,0 +1,1471 @@
...
+   /vllm-workspace/benchmarks/benchmark_throughput.py --dataset
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+   ROCR_VISIBLE_DEVICES="6" python3
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:624]:
```suggestion
   CUDA_VISIBLE_DEVICES="6" python3
```

File: docs/how-to/tuning-guides/mi300x.rst [hunk 9]
```diff
@@ -0,0 +1,1471 @@
...
+   /vllm-workspace/benchmarks/benchmark_throughput.py --dataset
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+   ROCR_VISIBLE_DEVICES="7" python3
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:629]:
```suggestion
   CUDA_VISIBLE_DEVICES="7" python3
```

File: docs/how-to/tuning-guides/mi300x.rst [hunk 10]
```diff
@@ -0,0 +1,1471 @@
...
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+Run two instances of ``llama3-8b`` model at the same time on one single GPU
+by specifying ``--gpu-memory-utilization`` to 0.4 (40%), as below (on GPU
+0):
+
+.. code-block:: shell
+
+   ROCR_VISIBLE_DEVICES=0 python3
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:640]:
```suggestion
   CUDA_VISIBLE_DEVICES=0 python3
```

File: docs/how-to/tuning-guides/mi300x.rst [hunk 11]
```diff
@@ -0,0 +1,1471 @@
...
+   /vllm-workspace/benchmarks/benchmark_throughput.py --gpu-memory-utilization
+   0.4 --dataset
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+   ROCR_VISIBLE_DEVICES=0 python3
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:646]:
```suggestion
   CUDA_VISIBLE_DEVICES=0 python3
```

File: docs/how-to/tuning-guides/mi300x.rst [hunk 12]
```diff
@@ -0,0 +1,1471 @@
...
+   /vllm-workspace/benchmarks/benchmark_throughput.py --gpu-memory-utilization
+   0.4 --dataset
+   "/path/to/dataset/ShareGPT_V3_unfiltered_cleaned_split.json" --model
+   /path/to/model &
+
+Similarly, use the ``ROCR_VISIBLE_DEVICES`` environment variable to specify
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:652]:
```suggestion
Similarly, use the ``CUDA_VISIBLE_DEVICES`` environment variable to specify
```

File: docs/how-to/tuning-guides/mi300x.rst [hunk 13]
```diff
@@ -0,0 +1,1471 @@
...
+which GPU (0-7) will run those instances. For example, the setting
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:653]:
```suggestion
which GPU (0-7) will run those instances.
```

File: docs/how-to/tuning-guides/mi300x.rst [hunk 14]
```diff
@@ -0,0 +1,1471 @@
...
+``ROCR_VISIBLE_DEVICES="4,5,6,7"`` exposes GPUs 4, 5, 6, and 7 to the program.
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:654]:
```suggestion
```

File: docs/how-to/tuning-guides/mi300x.rst [hunk 15]
```diff
@@ -0,0 +1,1471 @@
...
+Inside vLLM, the GPUs are mapped to ``CUDA_VISIBLE_DEVICES`` as 0, 1, 2, and 3.
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:655]:
```suggestion
```

File: docs/how-to/tuning-guides/mi300x.rst [hunk 16]
```diff
@@ -0,0 +1,1471 @@
...
+The two main reasons to use multiple GPUs:
+
+*  The model size is too big to run vLLM using one GPU as it results
+   CUDA/HIP Out of Memory.
+
+*  To achieve better latency.
+
+To run one vLLM instance on multiple GPUs, use the ``-tp`` or
+``--tensor-parallel-size`` option to specify multiple GPUs. Optionally, use the
+``ROCR_VISIBLE_DEVICES`` environment variable to specify the GPUs.
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:669]:
```suggestion
``CUDA_VISIBLE_DEVICES`` environment variable to specify the GPUs.
```

File: docs/how-to/tuning-guides/mi300x.rst [hunk 17]
```diff
@@ -0,0 +1,1471 @@
...
+below:
+
+.. code-block:: shell
+
+   python -m vllm.entrypoints.api_server --model /path/to/model --dtype
+   float16 -tp 2 --port 8000 &
+
+To achieve both latency and throughput performance for serving, you can
+run multiple API servers on different GPUs by specifying different ports
+for each server and use ``ROCR_VISIBLE_DEVICES`` to specify the GPUs for
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:681]:
```suggestion
for each server and use ``CUDA_VISIBLE_DEVICES`` to specify the GPUs for
```

File: docs/how-to/tuning-guides/mi300x.rst [hunk 18]
```diff
@@ -0,0 +1,1471 @@
...
+each server, for example:
+
+.. code-block:: shell
+
+   ROCR_VISIBLE_DEVICES=0,1 python -m vllm.entrypoints.api_server --model
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:686]:
```suggestion
   CUDA_VISIBLE_DEVICES=0,1 python -m vllm.entrypoints.api_server --model
```

File: docs/how-to/tuning-guides/mi300x.rst [hunk 19]
```diff
@@ -0,0 +1,1471 @@
...
+   /path/to/model --dtype float16 -tp 2 --port 8000 &
+
+   ROCR_VISIBLE_DEVICES=2,3 python -m vllm.entrypoints.api_server --model
```
Comment (hongxiayang) [inline docs/how-to/tuning-guides/mi300x.rst:689]:
```suggestion
   CUDA_VISIBLE_DEVICES=2,3 python -m vllm.entrypoints.api_server --model
```

**Review by hongxiayang — COMMENTED** [review]
Leaving some suggestions to update the environment variables used in vllm as vllm has been updated.

**Comment by peterjunpark** [comment]
> Leaving some suggestions to update the environment variables used in vllm as vllm has been updated.

@hongxiayang thanks for the updates

**Pull request closed without merging**